File: starters/JavaScript/bomberjam/logger.js
const fs = require('fs');

const format = (args) => {
  return args.map(x => typeof x !== 'string' ? x.toString() : x).join(' ');
}

class Logger {
  constructor() {
    this.file = null;
  }

  setup(filename) {
    if (!this.file)
      this.file = fs.createWriteStream(filename, {flags: 'w'});
  }

  debug(...args) {
    if (this.file)
      this.file.write(`DEBUG: ${format(args)}\n`);
  }

  info(...args) {
    if (this.file)
      this.file.write(`INFO: ${format(args)}\n`);
  }

  warn(...args) {
    if (this.file)
      this.file.write(`WARN: ${format(args)}\n`);
  }

  error(...args) {
    if (this.file)
      this.file.write(`ERROR: ${format(args)}\n`);
  }

  close() {
    if (this.file) {
      this.file.end();
      this.file = null;
    }
  }
}

module.exports = Logger;

File: starters/Python3/bomberjam/logger.py
import logging


class Logger:
    def __init__(self):
        self._is_configured = False

    def setup(self, filename):
        if not self._is_configured:
            logging.basicConfig(filename=filename, level=logging.DEBUG, format='%(levelname)s: %(message)s')
            self._is_configured = True

    def debug(self, text):
        if self._is_configured:
            logging.debug(text)

    def info(self, text):
        if self._is_configured:
            logging.info(text)

    def warn(self, text):
        if self._is_configured:
            logging.warning(text)

    def error(self, text):
        if self._is_configured:
            logging.error(text)
